kivy/uix/recyclelayout.py

```diff
@@ -22,19 +24,42 @@ class RecycleLayout(RecycleLayoutManagerBehavior, Layout):
     RecycleLayout provides the default layout for RecycleViews.
     """
 
-    default_size = ObjectProperty((100, 100))
+    default_width = NumericProperty(100)
+    default_height = NumericProperty(100)
```

Comment:
Hi, I think this part breaks the API if `default_width` and `default_height` don't accept None.

Reply:
Indeed, from what i gather, it only makes a difference by falling back to initial_size (which can't be None, None), but that would indeed be an api break (as the doc line after that says).

Reply:
I don't know how `initial_size` works, but I know setting `default_size` to `(None, None)` is useful in some situation e.g. [this](https://gist.github.com/gottadiveintopython/6fe590467cd62391084dce650097e757). If you comment out the `default_size: None, None` in that file, and resize the window or scroll the RecycleView while it's running, sometimes you'll see  the layout will be messed up as follows:

![](https://qiita-user-contents.imgix.net/https%3A%2F%2Fqiita-image-store.s3.ap-northeast-1.amazonaws.com%2F0%2F144673%2F5b935d13-2d74-0bd4-f5f9-f3676010aca0.png?ixlib=rb-1.2.2&auto=format&gif-q=60&q=75&s=e6ff953dbfb63b9ccc8aa23f4b70be51)

So I believe it's not just "fall back", and has another meaning.

Reply:
@gottadiveintopython @tshirtman was correct, it's only used as a fallback. The reason why you notice it having an effect elsewhere is likely because that example is using it in a buggy way. I.e. `height: self.texture_size[1]` is not reliable because it's not re-triggered when the view is reused.

Reply:
Yes, `initial_size` itself may exist just for a fallback, but I feel like setting `default_size` to `(None, None)` has more effect besides that.

> that example is using it in a buggy way

Well I couldn't find another way to do it (at least in a simple way). If there is one, could you teach me how to do it? (If you don't mind)

Reply:
A super hacky way to do that is add `on_y: self.height = self.texture_size[1]`. But that's just meant to illustrate what's going wrong.

The canonical way of doing it as follows:
```py
class ChatMsg(RecycleDataViewBehavior, Label):

    def refresh_view_layout(self, *args, **kwargs):
        super().refresh_view_layout(*args, **kwargs)
        self.height = self.texture_size[1]
```

and change the kv to not create the class. Anything else is just not safe.

Reply:
Thank you :) I'll invest that method.